torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -1837,6 +1904,26 @@ def forward(ctx, *deduped_flat_tensor_args):
                     assert isinstance(bw_state, BackwardState)
                     ctx._compiled_autograd_backward_state = bw_state
 
+                if num_rng:
+                    if len(fwd_rng_states) == 0:
+                        assert graphsafe_idx is not None
+                        initialize_rng_states(
+                            num_rng, graphsafe_idx, fwd_rng_states, bwd_rng_states
+                        )
+
+                    _curr_iter = next(curr_fwd_iter)
+                    ctx._curr_iter = _curr_iter
+
+                    # if this state is not contained in the backward,
+                    # we need to save it for when its backward pass happens
+                    if _curr_iter != backward_state_position:
+                        saved_backward_tensor_states[_curr_iter] = [
+                            rng_state.get_state() for rng_state in fwd_rng_states
+                        ]
+
+                    pending_forwards.add(_curr_iter)
+                    args = (*args, *fwd_rng_states)
```

Comment:
fwiw, this will probably increase runtime overhead a bit now in cases where there are any rng ops, since we need to allocate a tuple / unpack the old args tuple. We could potentially speed this up if `args` here is a list, so you can just append the extra elements (I'm not too opinionated, but in general runtime overhead has been a concern)

Reply:
yea, i tried to extend at first, but i'm not aware how to make this a list. this is the input to autograd.Function.